jni/src/jni_util.cpp

```diff
@@ -203,6 +202,35 @@ std::string knn_jni::JNIUtil::ConvertJavaStringToCppString(JNIEnv * env, jstring
     return cppString;
 }
 
+knn_jni::QuantizationLevel knn_jni::JNIUtil::ConvertJavaStringToQuantizationLevel(JNIEnv * env, jobject javaString) {
+    if (javaString == nullptr) {
+        throw std::runtime_error("String cannot be null");
+    }
+
+    const char *cString = env->GetStringUTFChars((jstring) javaString, nullptr);
+    if (cString == nullptr) {
+        this->HasExceptionInStack(env, "Unable to convert java string to cpp string");
+
+        // Will only reach here if there is no exception in the stack, but the call failed
+        throw std::runtime_error("Unable to convert java string to cpp string");
+    }
+    std::string cppString(cString);
+    env->ReleaseStringUTFChars((jstring) javaString, cString);
+
+    if (cppString == "ScalarQuantizationParams_1") {
```

Comment:
Can we reduce the size of this string? Also, can we put the const in a variable?

Reply:
This is the value returned from ScalarQuantizationParams.getTypeIdentifier (`SegmentLevelQuantizationInfo.getQuantizationParams().getTypeIdentifier()`). 

The objective of this helper is to in effect change the Java `ScalarQuantizationParams.getTypeIdentifier` value to a C++ enum. I can put the constant in a variable though.

Do you think I should pass a different value from Java to this index loading parameter?